py/sync_examples.py

```diff
@@ -62,15 +91,82 @@ def write_file(p: str, txt: str) -> str:
 
 
 def strip_comment(line: str) -> str:
-    return line.strip(" #")
+    """Returns the content of a line without '#' and ' ' characters
+
+    remove leading '#', but preserve '#' that is part of a tag
+    example:
+    >>> '# #hello '.strip('#').strip()
+    '#hello'
+    """
+    return line.strip('#').strip()
+
+
+def parse_tags(description: str) -> Tuple[str, List[str]]:
+    """Creates tags from description.
+
+    Accepts a description containing tags and returns a (new_description, tags) tuple.
+
+    The convention for tags:
+    1. Any valid twitter hashtag
+
+    For example, accept a description in any of the following forms
+
+    1. Use a checklist to group a set of related checkboxes. #form #checkbox #checklist
+
+    2. Use a checklist to group a set of related checkboxes.
+       #form #checkbox #checklist
+
+    3. Use a #checklist to group a set of related checkboxes.
+       #form #checkbox
+
+    and return
+    ('Use a checklist to group a set of related checkboxes.', ['checkbox', 'checklist', 'form']). The list of tags will
+    be sorted and all tags will be converted to lowercase.
+
+    Args:
+        description: Complete description of an example.
+    Returns:
+        A tuple of new_description and a sorted list of tags. new_description is created by removing the '#' characters
+        from the description.
+    """
+    hashtag_regex_pattern = r"(\s+)#(\w*[a-zA-Z]+\w*)\b"
+    pattern = re.compile(hashtag_regex_pattern)
+    matches = pattern.findall(' ' + description)
+
+    # Retrieve tags from the matches
+    tags = sorted(list(set([x[-1].lower() for x in matches])))
+
+    # Remove the '#' before the tags in description
+    new_d = pattern.sub(r'\1\2', ' ' + description)
+
+    # Remove the last line in description if it has only tags
```

Comment:
Not sure why this part is required. We could simply trim the description.

Reply:
@lo5 This is to make this example work.

```
Use a #checklist to group a set of related checkboxes. 
#form #checkbox
```

If the tags are part of the description, we don't want to remove the entire word, but only the `#` in front of it.

I trim the description, if the tags are at the end of it.